godtools/App/Share/Data/OptInOnboardingBannerEnabledRepository/OptInOnboardingBannerEnabledRepository.swift

```diff
@@ -7,16 +7,32 @@
 //
 
 import Foundation
+import Combine
 
 class OptInOnboardingBannerEnabledRepository {
+        
+    private static let sharedPublishers: OptInOnboardingBannerEnabledRepositoryPublishers = OptInOnboardingBannerEnabledRepositoryPublishers()
     
-    let userDefaultKey = UserDefaultKeys.optInOnboardingBannerDisabled
+    private let cache: OptInOnboardingBannerEnabledCache
+    private let publishers: OptInOnboardingBannerEnabledRepositoryPublishers
     
-    func getBannerIsEnabled() -> Bool {
-        return UserDefaults.standard.optInOnboardingBannerDisabled == false
+    init(cache: OptInOnboardingBannerEnabledCache) {
+        
+        self.cache = cache
+        self.publishers = OptInOnboardingBannerEnabledRepository.sharedPublishers
+        
+        publishers.enabled.value = cache.getEnabled()
     }
     
-    func disableBanner() {
-        UserDefaults.standard.optInOnboardingBannerDisabled = true
+    func getEnabled() -> AnyPublisher<Bool?, Never> {
+                
+        return publishers.enabled.eraseToAnyPublisher()
+    }
+    
+    func storeEnabled(enabled: Bool) {
+        
+        cache.storeEnabled(enabled: enabled)
+        
+        publishers.enabled.send(enabled)
```

Comment:
My initial hope was that userDefaults would allow us to have a publisher on it directly, so that we didn't have to manually publish changes like this, they would just happen automatically when we store a new value in UserDefaults.  But it seems like UserDefaults itself doesn't have a way to support publishers.

Reply:
I would prefer this as well.  Maybe it's possible.  I was trying to get it to work originally and thought it wasn't working and then I switched to this setup, then I realized I wasn't doing Combine correctly in the UseCase.  I never got back to testing UserDefaults again.  I'm going to create a ticket to explore publishing those changes from UserDefaults.